server/controllers/finance/distributionFeeCenter/configuration.js

```diff
@@ -7,61 +7,60 @@
 */
 const referenceAccount = require('./referenceAccount');
 const generalLedger = require('../generalLedger');
+const fiscal = require('../fiscal');
 
-function configuration(req, res, next) {
-  const { query } = req;
-  let refAccounts;
+async function configuration(req, res, next) {
+  try {
+    const { query } = req;
+    const params = {
+      typeFeeCenter : query.typeFeeCenter,
+      fee_center_id : query.fee_center_id,
+    };
 
-  const params = {
-    typeFeeCenter : query.typeFeeCenter,
-    fee_center_id : query.fee_center_id,
-  };
+    const accounts = await referenceAccount.auxilliary(params);
+    const refAccounts = accounts;
+    const accountsId = accounts.map(account => account.account_id);
 
-  referenceAccount.auxilliary(params)
-    .then((accounts) => {
-      if (!accounts.length) {
-        return [];
-      }
+    const options = {
+      accounts_id : accountsId,
+      excludes_distributed : true,
+      account_id : query.account_id,
+      trans_id : query.trans_id,
+      hrRecord : query.hrRecord,
+      limit : 100,
+    };
 
-      refAccounts = accounts;
-      const accountsId = accounts.map(account => account.account_id);
+    if (query.periodFrom && query.periodTo) {
```

Comment:
Do we need this check now?  I think this will turn our period IDs into dates, which are much less efficient to query in SQL.  Instead of changing them into dates, can you just add some more filters that will handle this?  Maybe a `filters.custom()` or something?